### ISSUE TITLE
Warnings in `cargo doc` 

### ISSUE THREAD

**Issue by briansmith** [ctz/rustls#756]
See below. I think PR #749 resolves some of these.

```
warning: unresolved link to `process_new_packets`
   --> rustls\src\conn.rs:209:10
    |
209 |     /// [`process_new_packets()`] has been called, this might start to return `false` while the
    |          ^^^^^^^^^^^^^^^^^^^^^^^ no item named `process_new_packets` in scope
    |
    = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: public documentation for `ConfigBuilder` links to private item `ClientConfigBuilder`
  --> rustls\src\builder.rs:46:44
   |
46 | /// 5. Now see [`ServerConfigBuilder`] or [`ClientConfigBuilder`] for further steps.
   |                                            ^^^^^^^^^^^^^^^^^^^^^ this item is private
   |
   = note: `#[warn(rustdoc::private_intra_doc_links)]` on by default
   = note: this link will resolve properly if you pass `--document-private-items`

warning: public documentation for `ConfigBuilder` links to private item `ClientConfigBuilder`
  --> rustls\src\builder.rs:46:44
   |
46 | /// 5. Now see [`ServerConfigBuilder`] or [`ClientConfigBuilder`] for further steps.
   |                                            ^^^^^^^^^^^^^^^^^^^^^ this item is private
   |
   = note: this link will resolve properly if you pass `--document-private-items`

warning: unresolved link to `process_new_packets`
   --> rustls\src\conn.rs:209:10
    |
209 |     /// [`process_new_packets()`] has been called, this might start to return `false` while the
    |          ^^^^^^^^^^^^^^^^^^^^^^^ no item named `process_new_packets` in scope
    |
    = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`

warning: 4 warnings emitted
```

**Comment by briansmith** [comment]
It seems like we should have a `cargo doc` job with warnings-as-errors in CI to catch things like this.

**Comment by ctz** [comment]
Addressed in 4f79c333
